addons/google_calendar/models/calendar.py

```diff
@@ -209,6 +216,126 @@ def _odoo_reminders_commands(self, reminders=()):
+        """
+        update_dict = self._get_time_update_dict(self, time_values)
+        time_values.update(update_dict)
+
+        # Get base values from the previous recurrence.
+        [previous_recurrence_values] = self.recurrence_id.copy_data()
+
+        # Edge case: selected event is the base_event of current recurrence, save events before mass_archive.
+        is_base_event_update = (self == self.recurrence_id.base_event_id)
+        previous_recurrence_events = self.recurrence_id._get_events_from(self.start)
+        detached_events_split = self.env['calendar.event']
+
+        if is_base_event_update:
+            # Delete current recurrence and reactivate base_event to create a new one.
+            self.action_mass_archive('all_events')
+            self.write({'active': True, 'recurrence_id': False, 'need_sync': False})
+        else:
+            # Stop previous recurrence at current event.
+            detached_events_split |= self.recurrence_id._stop_at(self)
+            # Detach following events (except for the current event).
+            (detached_events_split - self).write({'google_id': False, 'active': False})
+
+        # Update current event with the new recurrence information.
+        if values or time_values:
+            self.write({**time_values, **values, **{'google_id': False, 'need_sync': False}})
+
+        # Concatenate the parameters for the new recurrence and create it.
+        start_date = time_values['start'].date() if 'start' in time_values else self.start.date()
+        new_values = {
+            **previous_recurrence_values,
+            **self._get_recurrence_params_by_date(start_date),
+            **recurrence_values
+        }
+        # Update new_values recurrence fields for creating the new recurrence.
+        if 'rrule' in new_values:
+            new_values.pop('rrule')
+        if 'count' in new_values:
+            if is_base_event_update:
+                new_values['count'] = recurrence_values.get('count', 0) or len(previous_recurrence_events)
+            else:
+                new_values['count'] = recurrence_values.get('count', 0) or len(detached_events_split)
+
+        # Generate the new recurrence.
+        self._apply_recurrence_values(new_values)
+
+        # Return an empty list, there are no events left for passing through unlink() in method from super.
+        detached_events_empty = self.env['calendar.event']
+        return detached_events_empty
+
+
+    def _rewrite_recurrence(self, values, time_values, recurrence_values):
+        """ Update all events from a recurrence. Time changes are allowed only within the same day.
+        When changing recurrence values, patch the recurrence and recreate events locally.
+        The base event is kept in all flows.
+        """
+        self.ensure_one()
+        base_event = self.recurrence_id.base_event_id
+        update_dict = self._get_time_update_dict(base_event, time_values)
+        time_values.update(update_dict)
+
```

Comment:
The method is quite similar to the _rewrite_recurrence method in the calendar module.
Could some part of the code be refactored? If we do some changes in calendar, they won't be available in google_calendar. Factorizing the common parts and call super would help having working code easier to fix in case of issues.